msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplicationConfiguration.kt

```diff
@@ -0,0 +1,214 @@
+            )
+        }
+
+        // If we are given a redirectUri, we should run the superclass validation method to make sure
+        // we can use web auth
+        if (redirectUri != null) {
+            super.validateConfiguration()
+        } else {
+            Logger.warn(TAG, "No redirect URI was passed.")
+        }
+
+        // Enforce that account mode must be "SINGLE"
+        if (accountMode != AccountMode.SINGLE) {
+            throw MsalClientException(
+                MsalClientException.NATIVE_AUTH_INVALID_ACCOUNT_MODE_CONFIG_ERROR_CODE,
+                MsalClientException.NATIVE_AUTH_INVALID_ACCOUNT_MODE_CONFIG_ERROR_MESSAGE
+            )
+        }
+
+        // We only allow one authority being passed
+        if (authorities == null || authorities.size == 0) {
+            // Throw is no authority passed
+            throw MsalClientException(
+                MsalClientException.NATIVE_AUTH_USE_WITH_NO_AUTHORITY_ERROR_CODE,
+                MsalClientException.NATIVE_AUTH_USE_WITH_NO_AUTHORITY_ERROR_MESSAGE
+            )
+        }
+        else if (authorities.size > 1)
+        {
+            // This throws when more than one authority is passed
+            throw MsalClientException(
+                MsalClientException.NATIVE_AUTH_USE_WITH_MULTI_AUTHORITY_ERROR_CODE,
+                MsalClientException.NATIVE_AUTH_USE_WITH_MULTI_AUTHORITY_ERROR_MESSAGE
+            )
+        }
+
+        if (defaultAuthority !is NativeAuthCIAMAuthority) {
+            // If the authority supplied is a CIAMAuthority, we can continue
+            if (defaultAuthority is CIAMAuthority) {
+                val nativeAuthAuthority = NativeAuthCIAMAuthority(
+                    authorityUrl = defaultAuthority.authorityUri.toString(),
+                    clientId = clientId
+                )
+                authorities.clear()
+                authorities.add(nativeAuthAuthority)
+            } else {
+                // If not, throw exception
+                throw MsalClientException(
+                    MsalClientException.NATIVE_AUTH_INVALID_CIAM_AUTHORITY_ERROR_CODE,
+                    MsalClientException.NATIVE_AUTH_INVALID_CIAM_AUTHORITY_ERROR_MESSAGE
+                )
+            }
+        }
+
+        // Enforce that shared device mode is disabled
+        if (isSharedDevice) {
+            throw MsalClientException(
+                MsalClientException.NATIVE_AUTH_SHARED_DEVICE_MODE_ERROR_CODE,
+                MsalClientException.NATIVE_AUTH_SHARED_DEVICE_MODE_ERROR_MESSAGE
+            )
+        }
+
+        // Enforce that dev cannot set "broker_redirect_uri_registered = true"
+        // This check is technically not necessary, since Native Auth configuration only loads with a
+        // CIAM authority, which does not result in creating a broker controller
+        // Adding it for clarity with devs
+        if (useBroker) {
```

Comment:
Checking this really doesn't matter because the default config for MSAL already has set this to true, and you want that if you want to fallback to Browser anyway. You may choose to entirely remove this check.